lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -246,6 +257,38 @@ public function post($uri, $params)
         $this->makeRequest("POST", $uri, $params);
     }
 
+    /**
+     * Make PUT request
+     *
+     * String type was added to parameter $param in order to support sending JSON or XML requests.
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     *
+     * @see \Magento\Framework\HTTP\Client#patch($uri, $params)
```

Comment:
That looks strange for me, I would remove it. This method does not exisits in Magento\Framework\HTTP\Client. Even in ClientInterface
Pls check other locations